terraform/permissions/main.tf

```diff
@@ -0,0 +1,251 @@
+  name      = "superpass"
+  plaintext = local.berglas_secrets_list.superpass
+}
+
+
+resource google_storage_object_access_control superpass_sa_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.superpass.name
+  role   = "READER"
+  entity = "user-${local.sa_email}"
+
+  depends_on = [berglas_secret.superpass]
+}
+
+
+resource google_storage_object_access_control superpass_sa_cb_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.superpass.name
+  role   = "READER"
+  entity = "user-${local.sa_cb_email}"
+
+  depends_on = [berglas_secret.superpass, google_storage_object_access_control.superpass_sa_email]
+}
+
+
+#
+# Defining secret "database_url"
+resource berglas_secret database_url {
+  bucket    = var.berglas_bucket
+  key       = local.berglas_key
+  name      = "database_url"
+  plaintext = local.berglas_secrets_list.database_url
+}
+
+
+resource google_storage_object_access_control database_url_sa_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.database_url.name
+  role   = "READER"
+  entity = "user-${local.sa_email}"
+
+  depends_on = [berglas_secret.database_url]
+}
+
+
+resource google_storage_object_access_control database_url_sa_cb_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.database_url.name
+  role   = "READER"
+  entity = "user-${local.sa_cb_email}"
+
+  depends_on = [berglas_secret.database_url, google_storage_object_access_control.database_url_sa_email]
+}
+
+
+#
+# Defining secret "secret_key"
+resource berglas_secret secret_key {
+  bucket    = var.berglas_bucket
+  key       = local.berglas_key
+  name      = "secret_key"
+  plaintext = local.berglas_secrets_list.secret_key
+}
+
+
+resource google_storage_object_access_control secret_key_sa_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.secret_key.name
+  role   = "READER"
+  entity = "user-${local.sa_email}"
+
+  depends_on = [berglas_secret.secret_key]
+}
+
+
+resource google_storage_object_access_control secret_key_sa_cb_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.secret_key.name
+  role   = "READER"
+  entity = "user-${local.sa_cb_email}"
+
+  depends_on = [berglas_secret.secret_key, google_storage_object_access_control.secret_key_sa_email]
+}
+
+# GENERATED SECTION END
+
+###################################################################################
+
+# IAM Permissions for service accounts. 
+
+data "google_project" "project" {
+  project_id = var.project
+}
+
+resource "google_project_iam_binding" "sa_runview" {
+  role = "roles/run.viewer"
+
+  members = ["serviceAccount:${local.sa_email}"]
+}
+
+resource "google_project_iam_binding" "cloudbuild" {
+  for_each = toset([
+    "run.admin", "iam.serviceAccountUser", "cloudsql.admin"
+  ])
+
+  role    = "roles/${each.key}"
+  members = ["serviceAccount:${local.sa_cb_email}"]
+}
+
+resource "google_project_iam_binding" "cloudrun" {
+  for_each = toset([
+    "run.admin", "cloudsql.client"
```

Comment:
Not sure if I'm following this correctly, are you giving the Cloud Run service account permission to administrate Cloud Run services?

Reply:
Ah, this is a bug. I'm defining a `for_each` but never using it. (See line 247)